partner_no_address_type/tests/test_res_partner.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestBasePartnerTwoLine(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        partner_model = cls.env["res.partner"]
+        partner = partner_model.create(
+            {"name": "Test Company Name", "company_type": "company"}
+        )
+        cls.child_partner_name = partner_model.create(
+            {
+                "name": "Test Partner Name",
+                "type": "invoice",
+                "parent_id": partner.id,
+            }
+        )
+        cls.child_partner_no_name = partner_model.create(
+            {
+                "name": "",
+                "type": "invoice",
+                "parent_id": partner.id,
+            }
+        )
+
+    def test_get_name(self):
+        # Partner with name.
+        name = self.child_partner_name.with_context(no_address_type=True).display_name
+        self.assertEqual(name, "Test Company Name, Test Partner Name")
+        # Partner without a name.
+        name = self.child_partner_no_name.with_context(
+            no_address_type=True
+        ).display_name
+        self.assertEqual(name, "Test Company Name")
```

Comment:
BTW: This is actually testing the context key we must have in Odoo, not the application of context key when rendering report, so I'm not sure it makes any sense to have that here.

Reply:
Let's keep the test until https://github.com/odoo/odoo/pull/185166 is merged . Otherwise, we don't know whether the module works or not.